p2d/p2d.py

```diff
@@ -61,32 +79,21 @@ def write_ini(probid, timelimit, color):
             f.writelines(map(lambda s: s + '\n', ini_content))
         logger.info(END_OF_SUBPROCESS)
 
-    def write_yaml(name, memlimit):
+    def write_yaml():
         logger.info('Add \'problem.yaml\':')
         yaml_file = f'{output_dir}/problem.yaml'
-        name = name.replace('"', r'\"')
-        yaml_content = f'name: "{name}"\nlimits:\n  memory: {memlimit}\n'
+        _name = name.replace('"', r'\"')
+        yaml_content = f'name: "{_name}"\nlimits:\n  memory: {memlimit}\n'
         logger.info(yaml_content)
+        logger.info(END_OF_SUBPROCESS)
         logger.info('Add output validator:')
-
-        def get_checker_md5(checker):
-            if checker is None:
-                return None
-            checker_source = checker.find('source')
-            checker_file = '{}/{}'.format(package_dir, checker_source.attrib['path'])
-            with open(checker_file, 'r', encoding='utf-8') as f:
-                file_md5 = hashlib.md5(f.read().replace('\r\n', '\n').encode('utf-8'))
-            return file_md5.hexdigest().lower()
-
-        checker_md5 = get_checker_md5(checker)
-        logger.info('Checker md5: {}'.format(checker_md5))
+        checker_name = checker.attrib['name']
```

Comment:
![image](https://user-images.githubusercontent.com/10959348/121721575-fd0ae200-cb16-11eb-8273-7210349b587e.png)

There's no `name` attribute for self-defined checker.

```xml
        <checker type="testlib">
            <source path="files/chk.cpp" type="cpp.g++17"/>
            <copy path="check.cpp" type="cpp.g++17"/>
            <testset>
                <test-count>4</test-count>
                <input-path-pattern>files/tests/checker-tests/%02d</input-path-pattern>
                <output-path-pattern>files/tests/checker-tests/%02d.o</output-path-pattern>
                <answer-path-pattern>files/tests/checker-tests/%02d.a</answer-path-pattern>
                <tests>
                    <test verdict="ok"/>
                    <test verdict="crashed"/>
                    <test verdict="wrong-answer"/>
                    <test verdict="ok"/>
                </tests>
            </testset>
        </checker>
```